src/common/libtaskmap/taskmap.c

```diff
@@ -457,7 +457,7 @@ const struct idset *taskmap_taskids (const struct taskmap *map, int nodeid)
     struct taskmap_block *block;
     struct idset *taskids;
 
-    if (!map || nodeid < 0 || taskmap_unknown (map)) {
+    if (!map || nodeid < -1 || taskmap_unknown (map)) {
```

Comment:
Would this be slightly more clear if we defined
```c
#define TASKMAP_NODEID_ALL (-1)
```
in the header for users?

Reply:
On second thought, if that convenience parameter is confusing I'll just drop the change to libtaskmap and move the functionality into `flux job`.

Reply:
Sorry, the -1 is not particularly confusing, so I probably shouldn't have suggested that!

Reply:
No, in retrospect that was a little kludgy and I didn't even add tests. Since there's only one use case, better to just do it by hand in `flux-job.c`. What I really want is a way to create a "filled" idset (combine idset_create() and idset_set_range() into a single call)